sway/input/seatop_touch.c

```diff
@@ -0,0 +1,105 @@
+#define _POSIX_C_SOURCE 200809L
+#include <wlr/types/wlr_touch.h>
+#include <wlr/types/wlr_cursor.h>
+#include "sway/input/seat.h"
+#include "sway/input/cursor.h"
+#include "log.h"
+#include "config.h"
+
+struct seatop_touch_point_event {
+	double ref_lx, ref_ly;         // cursor's x/y at start of op
+	double ref_con_lx, ref_con_ly; // container's x/y at start of op
+	int32_t touch_id;
+	struct wl_list link;
+};
+
+struct seatop_touch_event {
+	struct wl_list point_events;
+};
+
+static void handle_touch_down(struct sway_seat *seat,
+		struct wlr_event_touch_down *event) {
+
+	struct sway_cursor *cursor = seat->cursor;
+	struct wlr_surface *surface = NULL;
+	struct seatop_touch_event *e = seat->seatop_data;
+
+	double lx, ly;
+	wlr_cursor_absolute_to_layout_coords(cursor->cursor, event->device,
+			event->x, event->y, &lx, &ly);
+	double sx, sy;
+	node_at_coords(seat, lx, ly, &surface, &sx, &sy);
+
+	if (!surface) { // Must start from the surface
+		return;
+	}
+
+	struct seatop_touch_point_event *point_event = calloc(1, sizeof(struct seatop_touch_point_event));
+	point_event->ref_lx = lx;
+	point_event->ref_ly = ly;
+	point_event->ref_con_lx = sx;
+	point_event->ref_con_ly = sy;
+	point_event->touch_id = event->touch_id;
+
+	wl_list_insert(&e->point_events, &point_event->link);
```

Comment:
We should handle sending `wlr_seat_touch_notify_down` here instead of in `cursor.c` now.

Reply:
This one too